doc/docs/master/deploy-manage/manage/backup_restore.md

```diff
@@ -1,199 +1,295 @@
+   kubectl get svc/dash -o json >dash_svc_backup_30080.json
+   ```
+
+   1. Switch the `etcd` and dashboard manifests:
+
+      ```bash
+      kubectl get svc/pachd -o json | sed 's/30651/30648/g' | kubectl apply -f -
+      kubectl get svc/pachd -o json | sed 's/30652/30647/g' | kubectl apply -f -
+      kubectl get svc/pachd -o json | sed 's/30654/30646/g' | kubectl apply -f -
+      kubectl get svc/pachd -o json | sed 's/30655/30644/g' | kubectl apply -f -
+      kubectl get svc/etcd -o json | sed 's/32379/32378/g' | kubectl apply -f -
+      kubectl get svc/dash -o json | sed 's/30080/30079/g' | kubectl apply -f -
+      kubectl get svc/dash -o json | sed 's/30081/30078/g' | kubectl apply -f -
+      kubectl get svc/pachd -o json | sed 's/30600/30611/g' | kubectl apply -f -
+      ```
+
+1. Modify your environment so that you can access `pachd` on this new
+port
+
+   ```bash
+   pachctl config update context `pachctl config get active-context` --pachd-address=<cluster ip>:30649
+   ```
+
+1. Verify that you can talk to `pachd`: (if not, firewall rules are a common culprit)
+
+   ```bash
+   pachctl version
+   ```
+
+   **System Response:**
+
+   ```
+   COMPONENT           VERSION
+   pachctl             1.10.0
+   pachd               1.10.0
+   ```
 
-Pachyderm provides the `pachctl extract` and `pachctl restore` commands to backup and restore the state of a Pachyderm cluster.
+??? note "pause-pipelines.sh"
+    Alternatively, you can run **Steps 1 - 3** by using the following script:
 
-The `pachctl extract` command requires that all pipeline and data loading activity into Pachyderm stop before the extract occurs.  This enables Pachyderm to create a consistent, point-in-time backup.  In this document, we'll talk about how to create such a backup and restore it to another Pachyderm instance.
+    ```bash
+    #!/bin/bash
+    # Stop all pipelines:
+    pachctl list pipeline --raw \
+    | jq -r '.pipeline.name' \
+    | xargs -P3 -n1 -I{} pachctl stop pipeline {}
 
-Extract and restore commands are currently used to migrate between minor and major releases of Pachyderm, so it's important to understand how to perform them properly.   In addition, there are a few design points and operational techniques that data engineers should take into consideration when creating complex pachyderm deployments to minimize disruptions to production pipelines.
+    # Backup the Pachyderm service spec, in case you need to restore it quickly
+    kubectl get svc/pachd -o json >pach_service_backup_30650.json
+    kubectl get svc/etcd -o json >etcd_svc_backup_32379.json
+    kubectl get svc/dash -o json >dash_svc_backup_30080.json
 
-In this document, we'll take you through the steps to backup and restore a cluster, migrate an existing cluster to a newer minor or major release, and elaborate on some of those design and operations considerations.
+    # Modify the service to accept traffic on port 30649
```

Comment:
We should add comments for each line, detailing what it does.

   # We must modify all the ports of all the Pachyderm services to avoid collisions with the migration cluster
   # modify the pachd API endpoint to run on 30649
    kubectl get svc/pachd -o json | sed 's/30650/30649/g' | kubectl apply -f -
   # modify the pachd trace port to run on 30648
    kubectl get svc/pachd -o json | sed 's/30651/30648/g' | kubectl apply -f -
   # modify the pachd api-over-http port to run on 30647
    kubectl get svc/pachd -o json | sed 's/30652/30647/g' | kubectl apply -f -
   # modify the pachd saml authentication port to run on 30646
    kubectl get svc/pachd -o json | sed 's/30654/30646/g' | kubectl apply -f -
   # modify the pachd git api callback port to run on 30644
    kubectl get svc/pachd -o json | sed 's/30655/30644/g' | kubectl apply -f -
   # modify the etcd client port to run on 32378
    kubectl get svc/etcd -o json | sed 's/32379/32378/g' | kubectl apply -f -
   # modify the dashboard ports to run on 30079 and 30078, respectively
    kubectl get svc/dash -o json | sed 's/30080/30079/g' | kubectl apply -f -
    kubectl get svc/dash -o json | sed 's/30081/30078/g' | kubectl apply -f -
   # modify the pachd s3 port to run on 30611
    kubectl get svc/pachd -o json | sed 's/30600/30611/g' | kubectl apply -f -

Reply:
added